pyrokinetics/kinetics/KineticsReaderSCENE.py

```diff
@@ -0,0 +1,107 @@
+            psi_n = psi / psi[-1]
+
+            rho = kinetics_data["TGLF_RMIN"][::-1]
+            rho_func = InterpolatedUnivariateSpline(psi_n, rho)
+
+            # Determine electron data
+            electron_temp_data = kinetics_data["Te"][::-1]
+            electron_temp_func = InterpolatedUnivariateSpline(psi_n, electron_temp_data)
+
+            electron_density_data = kinetics_data["Ne"][::-1]
+            electron_density_func = InterpolatedUnivariateSpline(
+                psi_n, electron_density_data
+            )
+
+            electron_rotation_data = electron_temp_data * 0.0
+            electron_rotation_func = InterpolatedUnivariateSpline(
+                psi_n, electron_rotation_data
+            )
+
+            electron = Species(
+                species_type="electron",
+                charge=-1,
+                mass=electron_mass,
+                dens=electron_density_func,
+                temp=electron_temp_func,
+                rot=electron_rotation_func,
+                rho=rho_func,
+            )
+
+            # Determine ion data
+            ion_temperature_func = electron_temp_func
+            ion_rotation_func = electron_rotation_func
+
+            ion_density_func = InterpolatedUnivariateSpline(
+                psi_n, electron_density_data / 2
+            )
+
+            deuterium = Species(
+                species_type="deuterium",
+                charge=1,
+                mass=deuterium_mass,
+                dens=ion_density_func,
+                temp=ion_temperature_func,
+                rot=ion_rotation_func,
+                rho=rho_func,
+            )
+
+            tritium = Species(
+                species_type="tritium",
+                charge=1,
+                mass=1.5 * deuterium_mass,
+                dens=ion_density_func,
+                temp=ion_temperature_func,
+                rot=ion_rotation_func,
+                rho=rho_func,
+            )
+
+            # Return dict of species
+            return {
+                "electron": electron,
+                "deuterium": deuterium,
+                "tritium": tritium,
+            }
+
+    def verify(self, filename: Union[str, Path]) -> None:
+        """Quickly verify that we're looking at a SCENE file without processing"""
+        # Try opening data file
+        # If it doesn't exist or isn't netcdf, this will fail
+        try:
+            data = xr.open_dataset(filename)
+        except FileNotFoundError as e:
+            raise FileNotFoundError(
+                f"KineticsReaderSCENE could not find {filename}"
+            ) from e
+        except ValueError as e:
+            raise ValueError(
+                f"KineticsReaderSCENE must be provided a NetCDF, was given {filename}"
+            ) from e
+        # Given it is a netcdf, check it has the expected data_vars
+        try:
+            if not np.all(np.isin(["Psi", "TGLF_RMIN", "Te", "Ne"], data.data_vars)):
```

Comment:
@ZedThree It might be worth adding in a description to the SCENE netCDF that identifies the SCENE version in output. If that gets added in then we can have a more rigorous check here. Not a pyrokinetics issue though

Reply:
Already done! https://github.com/PlasmaFAIR/SCENE/blob/d78c3d6861c7eed7854590ba52c128f1d393cb96/src/outcdf.f90#L224-L227

There are now `software_name` and `software_version` file-level attributes that could be checked.

It would be good if we could also try to get similar metadata into TRANSP and JETTO, but that might be a little bit harder

Reply:
TRANSP already has a version number which pyrokinetics now check if it exists, though it doesn't do anything with it.

JETTO has a build date but not any version info. I can probably add in a version output to the JETTO CDF at some point (sometimes it's good to fingers in a lot of pies!)

Reply:
I don't have access to a SCENE file with that metadata. What would be the best way to access `software_name` from the xarray dataset? Would it be as simple as checking `"SCENE" in data.software_name` and catching an AttributeError?

Reply:
Yes, I think that's right. I'll send you an example file